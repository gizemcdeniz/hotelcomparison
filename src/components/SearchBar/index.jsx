import React, { useState } from "react";
import { Row, Col, Container } from "react-bootstrap";
import searchIcon from "./images/searchIcon.svg";
import "./index.scss";

// This component takes in a function 'handleSearch' as a props, which handles search based on a keyword provided by user
const SearchBar = ({ handleSearch }) => {
  const [keyword, setKeyword] = useState("");

  const handleInputChange = (e) => {
    setKeyword(e.target.value);
  };

  return (
    <Container>
      <Row>
        <Col>
          <input
            placeholder="Search"
            className="mr-sm-2 searchBarInput"
            onChange={(e) => handleInputChange(e)}
            onKeyPress={(e) => {
              if (e.key === "Enter") {
                handleSearch(e, keyword);
                setKeyword("");
              }
            }}
          />
          <img
            className="searchIcon"
            src={searchIcon}
            alt="Search Icon"
            onClick={(e) => {
              handleSearch(e, keyword);
              setKeyword("");
            }}
          />
        </Col>
      </Row>
    </Container>
  );
};

export default SearchBar;
