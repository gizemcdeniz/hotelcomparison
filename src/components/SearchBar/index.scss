@import "../../style/variables";
@import "../../style/forms";

.searchBarInput {
  @include inputElement(
    $bgColor: $primaryColor,
    $placeholderColor: $primaryColor
  );
  height: 2.4rem;
  background-color: rgba(254, 143, 124, 0.3);
}

.searchIcon {
  position: absolute;
  right: 1rem;
  top: 20%;
  right: 2rem;
  width: 1.8rem;
  height: 1.8rem;
  z-index: 3;

  &:hover {
    cursor: pointer;
  }
}
