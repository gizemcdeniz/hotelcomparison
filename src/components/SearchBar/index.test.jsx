import React from "react";
import renderer from "react-test-renderer";
import SearchBar from "./index";

it("renders correctly and matches the snapshot", () => {
  const tree = renderer.create(<SearchBar />).toJSON();
  expect(tree).toMatchSnapshot();
});
