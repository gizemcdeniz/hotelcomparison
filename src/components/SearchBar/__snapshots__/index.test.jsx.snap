// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly and matches the snapshot 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <input
        className="mr-sm-2 searchBarInput"
        onChange={[Function]}
        onKeyPress={[Function]}
        placeholder="Search"
      />
      <img
        alt="Search Icon"
        className="searchIcon"
        onClick={[Function]}
        src="searchIcon.svg"
      />
    </div>
  </div>
</div>
`;
